src/commands/time/modules/time_conversion.ts

```diff
@@ -0,0 +1,82 @@
+export interface TimeZone {
+  name: string;
+  value: {
+    name: string;
+    isPositive: boolean;
+    hours: number;
+    minutes: number;
+  }
+}
+
+export interface TimeConverter {
+  hours: number;
+  minutes: number;
+  timeZoneGiven: TimeZone;
+  timeZoneConvert: TimeZone
+}
+
+export enum DayDiff {
+  SameDay,
+  NextDay,
+  PrevDay
+}
+
+export interface TimeResponse {
+  hours: number;
+  minutes: number;
+  day: DayDiff;
+}
+
+export function ConvertTime(timeConverter: TimeConverter) {
+  const givenZoneVal = timeConverter.timeZoneGiven.value
+  const convertZoneVal = timeConverter.timeZoneConvert.value
+
+  const expressionHours = `${(givenZoneVal.isPositive) ? '+' : '-'} ${givenZoneVal.hours} ${(convertZoneVal.isPositive) ? '+' : '-'} ${convertZoneVal.hours}`
+  const expressionMinutes = `${(givenZoneVal.isPositive) ? '+' : '-'} ${givenZoneVal.minutes} ${(convertZoneVal.isPositive) ? '+' : '-'} ${convertZoneVal.minutes}`
+
+  const hoursDiff = eval(expressionHours)
+  const minutesDiff = eval(expressionMinutes)
+
+  const convertedHours = timeConverter.hours + hoursDiff
+  const connvertedMinutes = timeConverter.minutes + minutesDiff
```

Comment:
Here's a suggestion of how it can be done
```suggestion
   const givenGmtOffset = givenZoneVal.isPositive ? 1 : -1;
  const convertGmtOffset = convertZoneVal.isPositive ? 1 : -1;
  const hoursDifference = givenZoneVal.hours * givenGmtOffset - convertZoneVal.hours * convertGmtOffset;
  const minuteDifference = givenZoneVal.minutes * givenGmtOffset - convertZoneVal.minutes * convertGmtOffset;

  const convertedHours = timeConverter.hours - hoursDifference;
  const convertedMinutes = timeConverter.minutes - minuteDifference;
```

Reply:
Oh, i didnt think of this method, this is nice.
Thanks